components/nodemanager-service/pgdb/nodes.go

```diff
@@ -370,7 +369,15 @@ func (db *DB) BulkAddNodes(inNodes []*nodes.Node) (nodeIDs []string, err error)
 		if err != nil {
 			return nodeIDs, errors.Wrap(err, "BulkAddNodes unable to translate node to db struct")
 		}
-		for _, host := range inNode.GetTargetConfig().GetHosts() {
+
+		hosts := inNode.GetTargetConfig().GetHosts()
+		if inNode.GetTargetConfig().Host != "" {
+			hosts = append(hosts, inNode.GetTargetConfig().Host)
+		}
+		if len(hosts) < 1 {
+			return nodeIDs, errors.New("BulkAddNodes unable find TargetConfig hosts")
```

Comment:
This error message reads weirdly and also doesn't provide a lot of specific information about how to fix it. How about something like: `BulkAddNodes requests require host or hosts in TargetConfig`.

Reply:
Thank you, always happy to make messages clearer. Pushed change